basepage/models.py

```diff
@@ -22,18 +23,39 @@ class Movie(models.Model):
     gross_earnings = models.CharField(max_length=100)
 
 
-class Cinema(models.Model):
-    city = models.CharField(max_length=200)
-    address = models.CharField(max_length=200)
-    hall = models.CharField(max_length=200)
+class SeatModel(models.Model):
+    available = models.BooleanField()
     name = models.CharField(max_length=200)
+
+
+class CinemaHall(models.Model):
+    name = models.CharField(max_length=200)
+    seats = models.ForeignKey(SeatModel, on_delete=models.SET_NULL, null=True)
```

Comment:
I think you got this wrong.
The Seat model should have a ForeignKey reference to the Hall.
The Hall model should have a ForeignKey to the Cinema.
This might help: https://www.geeksforgeeks.org/python-relational-fields-in-django-models/